src/Microsoft.AspNetCore.SignalR.Protocols.MsgPack/Internal/Protocol/MessagePackHubProtocol.cs

```diff
@@ -303,6 +303,35 @@ public void WriteMessage(HubMessage message, IBufferWriter<byte> output)
             }
         }
 
+        public byte[] GetMessageBytes(HubMessage message)
+        {
+            var writer = MemoryBufferWriter.Get();
+
+            try
+            {
+                // Write message to a buffer so we can get its length
+                WriteMessageCore(message, writer);
+
+                var dataLength = writer.Length;
+                var prefixLength = BinaryMessageFormatter.LengthPrefixLength(writer.Length);
+
+                var array = new byte[dataLength + prefixLength];
+                var span = array.AsSpan();
+
+                // Write length then message to output
+                var written = BinaryMessageFormatter.WriteLengthPrefix(writer.Length, span);
+                Debug.Assert(written == prefixLength);
+                written = writer.CopyTo(span.Slice(prefixLength));
+                Debug.Assert(written == dataLength);
```

Comment:
```
Debug.Assert(writer.Length == dataLength);
```

And change the return value of  CopyTo to void.

Reply:
`writer.Length` is where I get `dataLength` from :)

However, also have an assert in CopyTo so can remove this one.